### PR TITLE
Add functionality to display game history from file

### PR DESCRIPTION
**Pull request by szeliga71**
as title

### CODE REVIEW

File: src/main/java/org/example/gameService/LoadGameHistory.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import java.io.*;
+import java.time.LocalDate;
+import java.util.*;
+
+public class LoadGameHistory {
+
+    private static TimeDateProvider timeDateProvider = new TimeDateProvider();
+    private static final String mainPath = "src/main/GameHistory";
+
+    public static void setTimeDateProvider(TimeDateProvider provider) {
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gameService/LoadGameHistory.java]: please delete

File: src/main/java/org/example/gameService/LoadGameHistory.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
+package org.example.gameService;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.util.*;
+
+public class LoadGameHistory {
+
+    private static TimeDateProvider timeDateProvider = new TimeDateProvider();
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gameService/LoadGameHistory.java]: make final

File: src/main/java/org/example/gameService/LoadGameHistory.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    private static final String mainPath = "src/main/GameHistory";
+
+    public static void setTimeDateProvider(TimeDateProvider provider) {
+        timeDateProvider = provider;
+    }
+
+    public static String allGameFiles(List<File> files) {
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gameService/LoadGameHistory.java]: getAllGameFiles

File: src/main/java/org/example/gameService/LoadGameHistory.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    public static List<File> loadHistoryFile(String pathToDirectory) {
+        File directory = new File(pathToDirectory);
+        List<File> filesInDirectory = new ArrayList<>();
+        if (directory.isDirectory()) {
+            filesInDirectory = Arrays.asList(Objects.requireNonNull(directory.listFiles()));
+            filesInDirectory.sort(Comparator.comparing(File::getName));
+        } else {
+            System.out.println("The path to directory or the directory is invalid");
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gameService/LoadGameHistory.java]: Maybe throw illegalArgumentException?

File: src/main/java/org/example/gameService/LoadGameHistory.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        } else {
+            throw new IllegalArgumentException("Please provide a valid argument");
+        }
+    }
+
+    public static File createFile(LocalDate date) {
+        return new File(createPathToFile(date));
+    }
+
+    public static String gameResultFromSpecifiedDate(File searchedFile) {
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gameService/LoadGameHistory.java]: get*

File: src/main/java/org/example/gameService/TimeDateProvider.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+                int day = Integer.parseInt(scanner.nextLine());
+
+                System.out.println("Please enter a month (1-12): ");
+                int month = Integer.parseInt(scanner.nextLine());
+
+                System.out.println("Please enter a year(e.g,2023): ");
+                int year = Integer.parseInt(scanner.nextLine());
+
+                date = LocalDate.of(year, month, day);
+                break;
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gameService/TimeDateProvider.java]: rather use return date

File: src/main/java/org/example/gameService/TimeDateProvider.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
+package org.example.gameService;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.Scanner;
+
+public class TimeDateProvider {
+
+    public LocalTime getCurrentTime() {
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gameService/TimeDateProvider.java]: static

File: src/main/java/org/example/gameService/TimeDateProvider.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        return LocalTime.now();
+    }
+
+    public LocalDate inputDate(Scanner scanner) {
```
Comment (blazejprzyluski) [inline src/main/java/org/example/gameService/TimeDateProvider.java]: static

File: src/test/java/LoadGameHistoryTest.java [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+import static org.mockito.Mockito.when;
+
+
+@ExtendWith(MockitoExtension.class)
+public class LoadGameHistoryTest {
+
+    @Mock
+    private Scanner scanner;
+    @Mock
+    TimeDateProvider mockTimeDateProvider;
```
Comment (blazejprzyluski) [inline src/test/java/LoadGameHistoryTest.java]: private

File: src/main/java/org/example/gameService/LoadGameHistory.java (hunk 1 above)
Comment (szeliga71) [reply src/main/java/org/example/gameService/LoadGameHistory.java]: done

File: src/main/java/org/example/gameService/LoadGameHistory.java (hunk 2 above)
Comment (szeliga71) [reply src/main/java/org/example/gameService/LoadGameHistory.java]: done

File: src/main/java/org/example/gameService/LoadGameHistory.java (hunk 3 above)
Comment (szeliga71) [reply src/main/java/org/example/gameService/LoadGameHistory.java]: done

File: src/main/java/org/example/gameService/LoadGameHistory.java (hunk 4 above)
Comment (szeliga71) [reply src/main/java/org/example/gameService/LoadGameHistory.java]: done

File: src/main/java/org/example/gameService/LoadGameHistory.java (hunk 5 above)
Comment (szeliga71) [reply src/main/java/org/example/gameService/LoadGameHistory.java]: ok

File: src/main/java/org/example/gameService/TimeDateProvider.java (hunk 6 above)
Comment (szeliga71) [reply src/main/java/org/example/gameService/TimeDateProvider.java]: done

File: src/main/java/org/example/gameService/TimeDateProvider.java (hunk 7 above)
Comment (szeliga71) [reply src/main/java/org/example/gameService/TimeDateProvider.java]: done

File: src/main/java/org/example/gameService/TimeDateProvider.java (hunk 8 above)
Comment (szeliga71) [reply src/main/java/org/example/gameService/TimeDateProvider.java]: done

File: src/test/java/LoadGameHistoryTest.java (hunk 9 above)
Comment (szeliga71) [reply src/test/java/LoadGameHistoryTest.java]: done

**Pull request merged**